### PR TITLE
Fix RobertaEmbeddings

### PR DESCRIPTION
**Pull request by DSKSD**
First of all, the original implementation can define `segment_embeddings` depending on `num_segments` argument. Actually, their model(Roberta) didn't use `segment_embeddings` because they found the effectiveness of `FULL/DOC SENTENCE` setting of inputs. 

And `position_embeddings` should use `padding_idx` to ignore padded inputs. Also the embedding matrix's size should be `padding_idx + max_seq_length + 1`. (e.g If `padding_idx=1` and `max_seq_length=512`, `maxtrix size = (1 + 512 + 1) = 514`.

Last, `position_ids` should be made by considering the previous feature. Below is simple test to make `position_ids` to reflect `padding_idx` of `input_ids`
```
input_ids = torch.randint(0,1000,(3,10))
padding_idx = 0

### dummy padded input
input_ids[:,-2] = padding_idx
input_ids[:,-1] = padding_idx
input_ids[0][-3] = padding_idx
input_ids[-1][-3] = padding_idx

input_ids
>>> tensor([[946, 783, 399, 951, 496, 400, 350,   0,   0,   0],
            [905, 445, 410, 406, 526,   1, 255, 811,   0,   0],
            [815, 669, 813, 708, 475, 232, 190,   0,   0,   0]])

```

```
mask = input_ids.ne(padding_idx).int()
position_ids = (torch.cumsum(mask, dim=1).type_as(mask) * mask).long() + padding_idx

position_ids
>>> tensor([[1, 2, 3, 4, 5, 6, 7, 0, 0, 0],
            [1, 2, 3, 4, 5, 6, 7, 8, 0, 0],
            [1, 2, 3, 4, 5, 6, 7, 0, 0, 0]])

```

### CODE REVIEW

**Comment by julien-c** [comment]
This looks reasonable to me but I'd need to take a better look at it.

Maybe @myleott do you have time to take a quick glance?

File: pytorch_transformers/modeling_roberta.py [hunk 1]
```diff
@@ -46,22 +46,46 @@
...
-class RobertaEmbeddings(BertEmbeddings):
+class RobertaEmbeddings(nn.Module):
     """
     Same as BertEmbeddings with a tiny tweak for positional embeddings indexing.
     """
     def __init__(self, config):
-        super(RobertaEmbeddings, self).__init__(config)
-        self.padding_idx = 1
+        super(RobertaEmbeddings, self).__init__()
+        self.padding_idx = config.padding_idx
```
Comment (myleott) [inline pytorch_transformers/modeling_roberta.py]: Just a reminder this value will need to be updated in the roberta config jsons

**Review by myleott — COMMENTED** [review]
Looks good to me

File: pytorch_transformers/modeling_bert.py [hunk 2]
```diff
@@ -214,6 +215,7 @@ def __init__(self,
             self.type_vocab_size = type_vocab_size
             self.initializer_range = initializer_range
             self.layer_norm_eps = layer_norm_eps
+            self.padding_idx = padding_idx
```
Comment (ikuyamada) [inline pytorch_transformers/modeling_bert.py]: If `padding_idx` is added to `BertConfig`, it should also be used in `BertEmbeddings`:
https://github.com/huggingface/pytorch-transformers/blob/e2a628a2233884921f08a225303e41fb4be58280/pytorch_transformers/modeling_bert.py#L249

File: pytorch_transformers/modeling_roberta.py (hunk 1 above)
Comment (DSKSD) [reply pytorch_transformers/modeling_roberta.py]: Thank you for your comment :) Yeah, It needs to be considered.

File: pytorch_transformers/modeling_bert.py (hunk 2 above)
Comment (DSKSD) [reply pytorch_transformers/modeling_bert.py]: Thank you!  I agree with you. This is more general for arbitrary vocab. (which has another index as padding)

File: pytorch_transformers/modeling_roberta.py [hunk 3]
```diff
@@ -46,22 +46,46 @@
...
+        self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size, 
+                                            padding_idx=config.padding_idx)
+        self.position_embeddings = nn.Embedding(self.padding_idx + config.max_position_embeddings + 1,
```
Comment (ikuyamada) [inline pytorch_transformers/modeling_roberta.py]: I think it is unnecessary to add `padding_idx` and `1` to `max_position_embeddings` because `max_position_embeddings` is already set to be 514 in the uploaded RoBERTa configuration.

```
In [1]: from pytorch_transformers import RobertaConfig
In [2]: RobertaConfig.from_pretrained('roberta-base').max_position_embeddings
Out[2]: 514```

Comment (DSKSD) [reply pytorch_transformers/modeling_roberta.py]: This is a definition issue. However, It is not general. 
In my case, I'm trying to train the model with my own vocab(I usually use 0 as `padding_idx`) and data.  And also I want to make it compatible with `max_seq_length`. In the RoBERTa's case, `514` doesn't mean it could receive actual `514` length input because of `padding_idx`.

Comment (tbright17) [inline pytorch_transformers/modeling_roberta.py]: `self.padding_idx + config.max_position_embeddings + 1 = 516`
will cause error when loading weights.
